### PR TITLE
Template update 3.0.1

### PR DESCRIPTION
**Pull request by nschcolnicov**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by nschcolnicov** [comment]
Had to update the .stat file snaps because after having to update the resource definitions for the pipeline, it affects the alignment results.

**Comment by nschcolnicov** [comment]
Besides pulling from the template, I also added a new config profile for setting the resources in CI tests, since I had to remove the max resources params and the checkmax function with the new template

**Comment by nschcolnicov** [comment]
nf-core linting seems to be failing due to an issue with the new template that is not picking up the igenomes_base value from the ignore params array. I have already reported this issue, I'll update the PR once this is fixed

**Comment by apeltzer** [comment]
Hm, not sure I find the issue on tools @nschcolnicov which ticket did you create?

**Comment by nschcolnicov** [comment]
The issue has already been fixed in dev, and will be released soon: https://nfcore.slack.com/archives/CE6PELWR4/p1728478362119159
On the meantime @atrigila fixed this issue temporarily by adding the ignore params to the nextflow_schema

**Comment by nschcolnicov** [comment]
@atrigila We will be merging this, while we wait for nf-core to merge the fix for the ignore params issue, for now we added the igenomes_base to the schema instead of ignoring it, we can simply remove it once its fixed

File: .github/workflows/template_version_comment.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  template_version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out pipeline code
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
+
+      - name: Read template version from .nf-core.yml
+        uses: pietrobolcato/action-read-yaml@1.0.0
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply .github/workflows/template_version_comment.yml:15]: You can also look at this for the working solution
https://github.com/nf-core/tools/pull/3212

**Comment by charles-plessy** [comment]
Sorry for the direct commit; I thought that it would materialize as a simple suggestion. I will check the GitHub docs in more details…

File: conf/base.config [hunk 2]
```diff
@@ -10,9 +10,10 @@
 
 process {
 
-    cpus   = { check_max( 1    * task.attempt, 'cpus'   ) }
-    memory = { check_max( 6.GB * task.attempt, 'memory' ) }
-    time   = { check_max( 4.h  * task.attempt, 'time'   ) }
+    // TODO nf-core: Check the defaults for all processes
```
Comment (charles-plessy) [inline conf/base.config:13]: I would just delete the TODO as you reviewed that point already when creating the pipeline.

File: conf/test_contamination.config [hunk 3]
```diff
@@ -14,11 +14,6 @@ params {
     config_profile_name        = 'Test profile'
     config_profile_description = 'Minimal test dataset to check pipeline function with contamination filter'
 
-    // Limit resources so that this can run on GitHub Actions
```
Comment (charles-plessy) [inline conf/test_contamination.config:17]: Don't you need the same changes as in `conf/test.config` ?

(same fore the other files below.)

**Review by charles-plessy — COMMENTED** [review]
Looks good to me; I only had minor comments.

**Review by apeltzer — APPROVED** [review]

**Pull request merged**